### ISSUE TITLE
[HOLD for payment 2023-06-19] [$250] Migrate IOUConfirmationList.js to function component

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#16162]
### Filenames
- [ ] [src/components/IOUConfirmationList.js](https://github.com/Expensify/App/blob/main/src/components/IOUConfirmationList.js)
    - **type:** `React.Component`
    - **has state values:** true
    - **has refs:** false
    - **has context:** false
    - **lifecycle methods:** none

### Task
- We currently have some class components in our codebase that we would like to refactor to a function component.
- Here's a link with some general advice on how to refactor a class component to a function component: https://react.dev/reference/react/Component#alternatives
- If you need additional guidance, please ask in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q)
- Test for any regressions and verify that there are no breaking changes

**Comment by luacmartins** [comment]
on hold

**Comment by marcaaron** [comment]
@luacmartins taking this issue off HOLD as per this [Slack post](https://expensify.slack.com/archives/C03TQ48KC/p1680708858739299)

**IMPORTANT:** If anything unexpected pops up during the refactor/review:
- [ ] Take it to [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) to get clarification on.
- [ ] Prefix the conversation with `[HOOK REFACTOR]` so that we can find it easily.
- [ ] Link the result of the conversation in this section of the Design Doc titled “[Best Practices Conversations](https://docs.google.com/document/d/1yVqzirMLdLTaoB68IYUJ0vNArj80IaFs9-5CVUFht9o/edit#bookmark=id.lgs1kmu1vo8s)”
- [ ] Add it to the STYLE.md guide in this [open PR](https://github.com/Expensify/App/pull/16882) so we can reference it and apply any teachings in the next phase

**Comment by luacmartins** [comment]
Cool! Thanks for letting me know. I’ll work on it some time next week.

**Comment by luacmartins** [comment]
We are refactoring this component as part of a manual requests issue, so I'll wait a bit to migrate it after we merge those changes.

**Comment by luacmartins** [comment]
No updates. I'm working on Manual Requests as my top priority, so won't get to this for a few days.

**Comment by luacmartins** [comment]
no updates yet. focused on manual requests

**Comment by luacmartins** [comment]
no updates yet. focused on manual requests which hopefully means I can get to this next week.

**Comment by luacmartins** [comment]
no updates yet. focused on manual requests which hopefully means I can get to this next week.

**Comment by luacmartins** [comment]
Gonna work on it this week.

**Comment by luacmartins** [comment]
Draft PR [here](https://github.com/Expensify/App/pull/19561)

**Comment by marcaaron** [comment]
Looks like this is in progress so just making this a Daily (I'm making all the issues daily)

**Comment by luacmartins** [comment]
Yea, it's in review

**Comment by luacmartins** [comment]
PR is merged. Gonna add the Bug label to get BZ to pay C+ for the review

**Comment by luacmartins** [comment]
@stephanieelliott we just need to pay @mananjadhav for their review on this PR - https://github.com/Expensify/App/pull/19561

**Comment by stephanieelliott** [comment]
Posted to Upwork and extended offer to @mananjadhav, please accept when you get a chance!

**Comment by stephanieelliott** [comment]
Reapplying the `Bug` label to get another BZ member to issue payment on this as I am OOO til July 30. Thanks @trjExpensify,  this issue is for C+ payment only and I have already hired in Upwork -- the only thing to do here is issue payment on the 19th!

**Comment by luacmartins** [comment]
Just waiting on payment

**Comment by trjExpensify** [comment]
>Just waiting on payment

Alright, alright.. I get the hint! ;) 

Settled up, closing!
